incubator/model-core/src/main/java/io/aklivity/zilla/runtime/model/core/internal/Int32ValidatorHandler.java

```diff
@@ -0,0 +1,126 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package io.aklivity.zilla.runtime.model.core.internal;
+
+import org.agrona.BitUtil;
+import org.agrona.DirectBuffer;
+
+import io.aklivity.zilla.runtime.engine.model.ValidatorHandler;
+import io.aklivity.zilla.runtime.engine.model.function.ValueConsumer;
+import io.aklivity.zilla.runtime.model.core.config.Int32ModelConfig;
+
+public class Int32ValidatorHandler implements ValidatorHandler
+{
+    private final int max;
+    private final int min;
+    private final int multiple;
+    private final boolean exclusiveMax;
+    private final boolean exclusiveMin;
+    private final IntegerFormat format;
+
+    private boolean number;
+    private boolean sign;
+    private int value;
+    private int progress;
+
+    public Int32ValidatorHandler(
+        Int32ModelConfig config)
+    {
+        this.max = config.max;
+        this.min = config.min;
+        this.exclusiveMax = config.exclusiveMax;
+        this.exclusiveMin = config.exclusiveMin;
+        this.multiple = config.multiple;
+        this.format = IntegerFormat.of(config.format);
+    }
+
+    @Override
+    public boolean validate(
+        int flags,
+        DirectBuffer data,
+        int index,
+        int length,
+        ValueConsumer next)
+    {
+        int position = 0;
+        boolean valid = true;
+        if ((flags & FLAGS_INIT) != 0x00)
+        {
+            value = 0;
+            progress = 0;
+            sign = false;
+            number = false;
+            byte digit = data.getByte(index);
+            if (format.negative(digit))
+            {
+                position += BitUtil.SIZE_OF_BYTE;
+                sign = true;
+            }
+        }
+
+        for (int numByteIndex = index + position; numByteIndex < index + length; numByteIndex++)
+        {
+            byte digit = data.getByte(numByteIndex);
+            if (format.digit(digit))
+            {
+                value = format.decode(value, digit);
+                number = true;
+            }
+            else
+            {
+                valid = false;
+                break;
+            }
+            progress += BitUtil.SIZE_OF_BYTE;
+        }
+
+        if (valid)
+        {
+            if ((flags & FLAGS_FIN) != 0x00 && format.validateFin(progress, number))
+            {
+                if (sign)
+                {
+                    value *= -1;
+                }
+                valid = conditions(value, max, min, exclusiveMax, exclusiveMin, multiple);
+            }
+            else
+            {
+                valid = format.validateContinue(progress);
+            }
+        }
+        return valid;
+    }
```

Comment:
If we use `MutableInteger`, then enum can modify state without inadvertently sharing state across validations or workers.

Using `decoded` field to store incrementally decoded result across multiple calls to `format.decode(...)`.
Using `processed` field to store total processed bytes across multiple calls to `format.decode(...)`.

```suggestion
    @Override
    public boolean validate(
        int flags,
        DirectBuffer data,
        int index,
        int length,
        ValueConsumer next)
    {
        if ((flags & FLAGS_INIT) != 0x00)
        {
            decoded.value = 0;
            processed.value = 0;
        }

        int progress = format.decode(decoded, processed, data, index, length);
        boolean valid = progress != IntegerFormat.INVALID_INDEX;

        if ((flags & FLAGS_FIN) != 0x00)
        {
            valid &= format.valid(decoded, processed);
            valid &= check.test(value);
        }

        return valid;
    }
```

where the enum becomes something like...
```
public enum IntegerFormat
{
    public static final int INVALID_INDEX = -1;

    TEXT
    {
        @Override
        public int decode(
            MutableInteger decoded,
            MutableInteger processed,
            DirectBuffer data,
            int index,
            int length)
        {
            int progress = index;
            int limit = index + length;

            decode:
            for (; progress < limit; progress++)
            {
                int digit = data.getByte(progress);

                if (digit < '0' || '9' < digit)
                {
                    if (processed.value == 0))
                    {
                        switch (digit)
                        {
                        case '-':
                            decoded.value = Integer.MIN_VALUE;
                            processed.value++;
                            continue decode;
                        case '+':
                            decoded.value = Integer.MAX_VALUE;
                            processed.value++;
                            continue decode;
                        default:
                            break;
                        }
                    }

                    progress = INVALID_INDEX;
                    break decode;
                }

                int multipler = 10;

                if (processed.value == 1))
                {
                    switch (decoded.value)
                    {
                    case Integer.MIN_VALUE:
                        decoded.value = -1;
                        multipler = 1;
                        break;
                    case Integer.MAX_VALUE:
                        decoded.value = 1;
                        multipler = 1;
                        break;
                    default:
                        break;
                    }
                }

                decoded.value = decoded.value * multipler + (digit - '0');
                processed.value++;
            }
        }

        @Override
        public boolean valid(
            MutableInteger decoded,
            MutableInteger processed)
        {
            return processed.value > 1 || decoded.value != Integer.MIN_VALUE || decoded.value != Integer.MAX_VALUE;
        }
    },

    BINARY
    {
        private static final int INT32_SIZE = 4;

        @Override
        public int decode(
            MutableInteger decoded,
            MutableInteger processed,
            DirectBuffer data,
            int index,
            int length)
        {
            int progress = index;
            int limit = index + length;

            decode:
            for (; progress < limit; progress++)
            {
                int digit = data.getByte(progress);

                if (processed.value >= INT32_SIZE)
                {
                    progress = INVALID_INDEX;
                    break decode;
                }

                decoded.value <<= 8;
                decoded.value |= digit & 0xFF;
                processed.value++;
            }

            return progress;
        }

        @Override
        public boolean valid(
            MutableInteger decoded,
            MutableInteger processed)
        {
            return processed.value == INT32_SIZE;
        }
    }
```